spec/01-unit/011-balancer_spec.lua

```diff
@@ -7,6 +7,9 @@ describe("Balancer", function()
 
 
   setup(function()
+    _G.ngx = ngx
+    _G.ngx.log = function() return end
```

Comment:
Maybe better to revert such global edits in the teardown hook?

Reply:
Replaced with a `stub`  (which to my understanding is lexically scoped, so it accomplishes the same task- can always revert the stub in teardown if you prefer).

Reply:
Had doubts about this (always had to undo the stubs manually), and confirmed: we should call `revert()` on the stub in the `teardown` hook

Reply:
Done! also used this same approach in the reports unit test, given that it was the basis for my original approach here